lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -411,6 +412,9 @@ std::error_code SerializedModuleLoader::findModuleFilesInDirectory(
   if (LoadMode == ModuleLoadingMode::OnlyInterface)
     return std::make_error_code(std::errc::not_supported);
 
+  if (ImplicitModulesDisabled)
+    return std::make_error_code(std::errc::not_supported);
```

Comment:
If we reposition `SerializedModuleLoader` to `ImplicitSerializedModuleLoader`, we don't need this any more.